jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryingLogGroups.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient
+import software.amazon.awssdk.services.cloudwatchlogs.model.StartQueryRequest
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+
+class QueryingLogGroups(private val project: Project) : CoroutineScope by ApplicationThreadPoolScope("ExecutingQuery") {
+    private var client: CloudWatchLogsClient = project.awsClient()
+    public fun executeStartQuery(qEndDate: Long, logGroupName: String, query: String, qStartDate: Long) = launch {
```

Comment:
Start and end need to be `Instant`'s

Reply:
Nit: might make more logical sense to have the start and end parameters next to each other, something like
```
public fun executeStartQuery(logGroupName: String, query: String, qStartDate: Instant, qEndDate: Instant)
```

Reply:
Converted start and end date to single object.